rcl/test/rcl/test_wait.cpp

```diff
@@ -553,3 +553,48 @@ TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), add_with_index) {
     EXPECT_EQ(&guard_conditions[i], wait_set.guard_conditions[i]);
   }
 }
+
+// Extra invalid arguments not tested
+TEST_F(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), wait_set_valid_arguments) {
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rcl_ret_t ret =
+    rcl_wait_set_init(&wait_set, 0, 0, 0, 0, 0, 0, context_ptr, rcl_get_default_allocator());
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  EXPECT_EQ(RCL_RET_WAIT_SET_EMPTY, rcl_wait(&wait_set, RCL_MS_TO_NS(1000)));
+  EXPECT_EQ(RCL_RET_OK, rcl_wait_set_fini(&wait_set)) << rcl_get_error_string().str;
+
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rcl_wait(nullptr, RCL_MS_TO_NS(1000)));
```

Comment:
@Blast545 here and elsewhere below, no `rcl_reset_error()` ?

Reply:
Still no `rcl_reset_error()`?

Reply:
I added `<< rcl_get_error_string().str` calls with the tests that are meant to fail. Doesn't this clean the rcl error state?

Reply:
Nope, just gets you the error string.

Reply:
My bad, I thought reading was enough to clean its state. Added rcl_reset_error calls in 7715e96.